wdl/src/main/scala/wdl/WdlNamespace.scala

```diff
@@ -457,13 +457,16 @@ object WdlNamespace {
     invalidVariableReferences ++ typeMismatches
   }
 
-  private def lookupType(from: Scope)(n: String): WomType = {
+  private [wdl] def lookupType(from: Scope)(n: String): WomType = {
     val resolved = from.resolveVariable(n)
     resolved match {
       case Some(d: DeclarationInterface) => d.relativeWdlType(from)
       case Some(c: WdlCall) => WdlCallOutputsObjectType(c)
       case Some(s: Scatter) => s.collection.evaluateType(lookupType(s), new WdlStandardLibraryFunctionsType, Option(from)) match {
-        case Success(a: WomArrayType) => a.memberType
+        case Success(WomArrayType(aType)) => aType
+        case Success(WomMapType(kType,vType)) =>
```

Comment:
This second line will never hit because of the custom unapply in `object WomArrayType`. I'd suggest removing it and adding a comment to that effect

Reply:
Fixed!